Assets/Scripts/Level/TutorialTrigger.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Assets.Scripts.Level
+{
+    /**
+     * Create an instance of TutorialTriggerGroups to populate each Trigger to speed up GetComponent calls
+     */
+    public class TutorialTrigger : MonoBehaviour
+    {
+        public int requiredDeaths;
+        public string[] TutorialText;
+        private PlayerProgress playerProgress;
+        private InGameUI gameUI;
+
+        private int initialDeaths;
+        private bool activated;
+
+        // Use this for initialization
+        void Start()
+        {
+            Collider myCollider = GetComponent<Collider>();
+            myCollider.isTrigger = true;
+            initialDeaths = -1;
+            activated = false;
+        }
+
+        public void AddGameUI(InGameUI ui)
+        {
+            gameUI = ui;
+        }
+
+        public void AddPlayerProgress(PlayerProgress player)
+        {
+            playerProgress = player;
+        }
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.gameObject.layer == PlayerConstants.PlayerLayer)
+            {
+                if (activated) return;
+                int currentDeaths = playerProgress.GetDeaths();
+                // On first trigger set initialDeaths = to current player deaths to allow us to 
+                // track the number of deaths experienced after first passing the trigger
+                // Also reset if player has reset level
+                if (initialDeaths == -1 || currentDeaths < initialDeaths)
+                {
+                    initialDeaths = currentDeaths;
+                }
+                if (currentDeaths - initialDeaths >= requiredDeaths)
+                {
+                    Debug.Log("Tutorial message generated");

```

Comment:
youll want to be sure to take this out before the PR is merged, otherwise this will get called a ton. We save the logs to a file, so if im trying to find errors, this could make it hard to wade through the data